src/proto_conversions.cc

```diff
@@ -310,4 +346,67 @@ bool RoadClass_Enum_Parse(const std::string& rc_name, valhalla::RoadClass* rc) {
   *rc = i->second;
   return true;
 }
+
+const std::unordered_map<int, std::string> vehicle_to_string{
+    {static_cast<int>(DirectionsLeg_VehicleType_kCar), "car"},
+    {static_cast<int>(DirectionsLeg_VehicleType_kMotorcycle), "motorcycle"},
+    {static_cast<int>(DirectionsLeg_VehicleType_kAutoBus), "bus"},
+    {static_cast<int>(DirectionsLeg_VehicleType_kTractorTrailer), "tractor_trailer"},
+    {static_cast<int>(DirectionsLeg_VehicleType_kMotorScooter), "motor_scooter"},
+};
+
+const std::unordered_map<int, std::string> pedestrian_to_string{
+    {static_cast<int>(DirectionsLeg_PedestrianType_kFoot), "foot"},
+    {static_cast<int>(DirectionsLeg_PedestrianType_kWheelchair), "wheelchair"},
+    {static_cast<int>(DirectionsLeg_PedestrianType_kSegway), "segway"},
+};
+
+const std::unordered_map<int, std::string> bicycle_to_string{
+    {static_cast<int>(DirectionsLeg_BicycleType_kRoad), "road"},
+    {static_cast<int>(DirectionsLeg_BicycleType_kCross), "cross"},
+    {static_cast<int>(DirectionsLeg_BicycleType_kHybrid), "hybrid"},
+    {static_cast<int>(DirectionsLeg_BicycleType_kMountain), "mountain"},
+};
+
+const std::unordered_map<int, std::string> transit_to_string{
```

Comment:
If these are `switch` statements, the compiler can help you remember to add new enum values.

Reply:
agree, I'd also choose `switch` or even `std::array` of size `DirectionsLeg_TransitType_TransitType_ARRAYSIZE`

Reply:
these were copy pasted from the serializer, i just moved them here because the belonged here organization wise. if we wnat to change the pattern that we have going we should do it for all of them at once. imho we should save that for another PR